Index: app/src/main/java/cn/edu/gdmec/android/draganddraw/BoxDrawingView.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/cn/edu/gdmec/android/draganddraw/BoxDrawingView.java	(revision 3a50f78cc87d5fee98ed70b080d826e38c5a001f)
+++ app/src/main/java/cn/edu/gdmec/android/draganddraw/BoxDrawingView.java	(revision )
@@ -1,18 +1,59 @@
 package cn.edu.gdmec.android.draganddraw;
 
 import android.content.Context;
+import android.graphics.PointF;
 import android.util.AttributeSet;
+import android.util.Log;
+import android.view.MotionEvent;
 import android.view.View;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Created by Jack on 2017/11/26.
  */
 
 public class BoxDrawingView extends View{
+    private static final String TAG = "BoxDrawingView";
+
+    private Box mCurrentBox;
+    private List<Box> mButton=new ArrayList<> (  );
+
     public BoxDrawingView(Context context){
         this(context, null);
     }
     public BoxDrawingView(Context context, AttributeSet attrs){
         super(context, attrs);
+    }
+    @Override
+    public boolean onTouchEvent(MotionEvent event){
+        PointF current = new PointF ( event.getX (), event.getY () );
+        String action = "";
+        switch (event.getAction ()){
+            case MotionEvent.ACTION_DOWN:
+                action = "ACTION_DOWN";
+                mCurrentBox = new Box ( current );
+                mButton.add ( mCurrentBox );
+                break;
+            case MotionEvent.ACTION_MOVE:
+                action = "ACTION_MOVE";
+                if (mCurrentBox != null){
+                    mCurrentBox.setCurrent ( current );
+                    invalidate ();
+                }
+                break;
+            case MotionEvent.ACTION_UP:
+                action = "ACTION_UP";
+                mCurrentBox = null;
+                break;
+            case MotionEvent.ACTION_CANCEL:
+                action = "ACTION_CANCEL";
+                mCurrentBox = null;
+                break;
+        }
+        Log.i ( TAG, action + " at x=" + current.x +
+        ", y=" + current.y);
+        return true;
     }
 }
Index: app/src/main/java/cn/edu/gdmec/android/draganddraw/Box.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/cn/edu/gdmec/android/draganddraw/Box.java	(revision )
+++ app/src/main/java/cn/edu/gdmec/android/draganddraw/Box.java	(revision )
@@ -0,0 +1,25 @@
+package cn.edu.gdmec.android.draganddraw;
+
+import android.graphics.PointF;
+
+/**
+ * Created by Jack on 2017/11/26.
+ */
+
+public class Box {
+    private PointF mOrigin;
+    private PointF mCurrent;
+    public Box(PointF origin){
+        mOrigin = origin;
+        mCurrent = origin;
+    }
+    public PointF getCurrent(){
+        return mCurrent;
+    }
+    public void setCurrent(PointF current){
+        mCurrent = current;
+    }
+    public PointF getOrigin(){
+        return mOrigin;
+    }
+}
